source/Private/New-GuestConfigurationPolicyContent.ps1

```diff
@@ -71,12 +75,41 @@ function New-GuestConfigurationPolicyContent
         Parameter = $Parameter
     }
 
+    if ($ContentManagedIdentity)
+    {
+        $metadataSectionParameters.ContentManagedIdentity = $ContentManagedIdentity
+    }
+
     $metadataSection = New-GuestConfigurationPolicyMetadataSection @metadataSectionParameters
 
     $parametersSection = New-GuestConfigurationPolicyParametersSection -Parameter $Parameter
 
     $conditionsSection = New-GuestConfigurationPolicyConditionsSection -Platform $Platform -Tag $Tag -IncludeVMSS $IncludeVMSS
 
+    if ($ContentManagedIdentity)
+    {
+        foreach ($anyOf in $conditionsSection.anyOf)
+        {
+            foreach ($allOf in $anyOf.allOf)
+            {
+                if ($allOf.value -eq "[parameters('IncludeArcMachines')]")
+                {
+                    # Find and remove the specified section
+                    $indexToRemove = $anyOf.allOf.IndexOf($allOf)
```

Comment:
Woah. This parameters silently removes the entire Arc section?
That wasn't anywhere in the description.
No where did it say that it was going to change the policy behind the scenes if I used this parameter.

This is where you need to write that message. And it should be a warning and not just a verbose message. But it should say something like "Using managed identity. Removing Arc machines from policy" or something.

But really I think maybe we need another boolean parameter to the cmdlet like "IncludeArcMachines" which is by default true or something.

Reply:
"Custom policies that reference the resource ID of a User Assigned Identity are not supported on Arc machines. The generated policy definition will not include Arc enabled servers to prevent execution."
Included this in the PR definition and is something that will be extensively documented.

Reply:
As discussed, included -ExcludeArcMachines flag that needs to be enabled for making user aware of the change.